source/AsyncRgbLedSimulationDataGenerator.cpp

```diff
@@ -15,57 +19,97 @@ AsyncRgbLedSimulationDataGenerator::~AsyncRgbLedSimulationDataGenerator()
 
 void AsyncRgbLedSimulationDataGenerator::Initialize( U32 simulation_sample_rate, AsyncRgbLedAnalyzerSettings* settings )
 {
+	// Initialize the random number generator with a literal seed to obtain repeatability
+    // Change this for srand(time(NULL)) for "truly" random sequences
+    // NOTICE rand() an srand() are *not* thread safe
+    srand( 42 );
+
 	mSimulationSampleRateHz = simulation_sample_rate;
 	mSettings = settings;
 
-	mSerialSimulationData.SetChannel( mSettings->mInputChannel );
-	mSerialSimulationData.SetSampleRate( simulation_sample_rate );
-	mSerialSimulationData.SetInitialBitState( BIT_HIGH );
+	mMaximumChannelValue = (1 << mSettings->BitSize()) - 1;
+
+    // TODO pass in the analyzer and call GetMinimumSampleRate?
+    mClockGenerator.Init( 12 * 1000000, mSimulationSampleRateHz );
+
+	mLEDSimulationData.SetChannel( mSettings->mInputChannel );
+	mLEDSimulationData.SetSampleRate( simulation_sample_rate );
+	mLEDSimulationData.SetInitialBitState( BIT_HIGH );
 }
 
 U32 AsyncRgbLedSimulationDataGenerator::GenerateSimulationData( U64 largest_sample_requested, U32 sample_rate, SimulationChannelDescriptor** simulation_channel )
 {
-	U64 adjusted_largest_sample_requested = AnalyzerHelpers::AdjustSimulationTargetSample( largest_sample_requested, sample_rate, mSimulationSampleRateHz );
+	U64 adjusted_largest_sample_requested = 
+	AnalyzerHelpers::AdjustSimulationTargetSample( largest_sample_requested, sample_rate, mSimulationSampleRateHz );
 
-	while( mSerialSimulationData.GetCurrentSampleNumber() < adjusted_largest_sample_requested )
+	while( mLEDSimulationData.GetCurrentSampleNumber() < adjusted_largest_sample_requested )
 	{
-		CreateSerialByte();
+        WriteReset();
```

Comment:
For testing purposes, we should toggle high speed mode every few cycles.
Since high speed mode isn't a setting of the analyzer, I recommend keeping it as a member variable of the simulation data generator, instead of a member of the settings class.

Reply:
Okay, I hadn't realised high-speed mode would vary like that, it's no problem to do that.

Reply:
Generally it won't in the real world, but it's a good way to exercise different features of the analyzer.